tests/unittest/auto_deploy/singlegpu/custom_ops/moe/test_torch_mxfp4_moe.py

```
@@ -0,0 +1,783 @@
+
+def test_mxfp4_transform_resolves_expert_block_size_and_rejects_unsupported() -> None:
+    layout = _deepseek_layout()
+
+    assert _resolve_mxfp4_expert_block_size({}, None) == _MXFP4_BLOCK_SIZE
+    assert _resolve_mxfp4_expert_block_size({}, layout) == _MXFP4_BLOCK_SIZE
+    assert (
+        _resolve_mxfp4_expert_block_size({"expert_block_size": _MXFP4_BLOCK_SIZE}, layout)
+        == _MXFP4_BLOCK_SIZE
+    )
+
+    with pytest.raises(ValueError, match="supports expert_block_size=32.*got 64"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)